src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/Content/PageBody/_details.html.twig

```diff
@@ -112,16 +114,16 @@
                                                     {% set currencyCode = channel.baseCurrency.code %}
                                                     {% set price = channelPricing.price %}
                                                     {% set originalPrice = channelPricing.originalPrice ?? 0 %}
-                                                    {% set lowestPriceBeforeDiscount = channelPricing.lowestPriceBeforeDiscount ?? 0 %}
+                                                    {% set lowestPriceBeforeDiscount = channelPricing.lowestPriceBeforeDiscount %}
 
-                                                    <tr>
-                                                        <td>
-                                                            <span class="badge me-2" style="background-color: {{ channel.color }};"></span>
+                                                    <tr {{ sylius_test_html_attribute('variants', "%s"|format(variant.name) ) }}>
+                                                        <td {{ sylius_test_html_attribute('channel-name') }}>
+                                                            <span class="badge me-2" style="background-color: {{ channel.color }};" ></span>
                                                             {{ channel.name }}
                                                         </td>
                                                         <td class="text-end">{{ _money.format(price, currencyCode) }}</td>
                                                         <td class="text-end">{{ _money.format(originalPrice, currencyCode) }}</td>
-                                                        <td class="text-end">{{ _money.format(lowestPriceBeforeDiscount, currencyCode) }}</td>
+                                                        <td class="text-end" {{ sylius_test_html_attribute('lowest-price-before-the-discount') }}>{% if lowestPriceBeforeDiscount != null %}{{ _money.format(lowestPriceBeforeDiscount, currencyCode) }}{% else %} - {% endif %}</td>
```

Comment:
Can we pass both the channel and variant's codes into the `lowest-price-before-the-discount` test attribute (join it with a dot, or just have 2 test attributes on the same element) and get the element directly instead of iterating over all visible variants just to check the price? Same deal w/ `price` and `originalPrice`.